### ISSUE TITLE
aria-allowed-attr not returning correct results for aria-roledescription

### ISSUE THREAD

**Issue by straker** [dequelabs/axe-core#1517]
Expectation: 
Aria-roledescription should pass for elements/roles listed in https://github.com/dequelabs/axe-core/issues/1216#issuecomment-454104190, return incomplete for all other roles, and fail for an element without a role

Actual: 
* Fails for `<a href="foobar.com" aria-roledescription="foo">foo</a>` (valid role but not supported on anchor elements, should incomplete)
* Fails for `<div role="button" aria-roledescription="foo">foo</button>` (valid role and supported on button elements/roles, should pass) 

Motivation: https://github.com/GoogleChrome/web.dev/issues/763

<pre><code>
axe-core version: 3.2.2
</code></pre>

**Comment by dylanb** [comment]
@straker Have you thought through the accessibility supported nuances here?

**Comment by straker** [comment]
Only in terms of the testing I did in https://github.com/dequelabs/axe-core/issues/1216#issuecomment-45410419

**Comment by jeankaplansky** [comment]
New rule help here: https://dequeuniversity.com/rules/axe/3.4/aria-roledescription

**Comment by chandana7393** [comment]
Tested, working fine.
![aria-roledescription](https://user-images.githubusercontent.com/45285854/67088316-788d8200-f1c2-11e9-9b1f-a856aba38f24.PNG)
**Tested Environment:**
Axe-coconut - 4.0.0.29104v
Chrome - 77.0.3865.120v
OS - Windows 10 64 bit.
